kie-server-parent/kie-server-tests/kie-server-integ-tests-jbpm/src/test/java/org/kie/server/integrationtests/jbpm/StartProcessServiceIntegrationTest.java

```diff
@@ -0,0 +1,129 @@
+
+        KieServerDeployer.buildAndDeployCommonMavenParent();
+        KieServerDeployer.buildAndDeployMavenProjectFromResource("/kjars-sources/restart-project");
+
+        kieContainer = KieServices.Factory.get().newKieContainer(releaseId);
+
+        createContainer(CONTAINER_ID_RESTART, releaseId, PPI_RUNTIME_STRATEGY);
+    }
+
+
+    @Test()
+    public void testStartProcess() {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID_RESTART, PROCESS_ID_RESTART, parameters);
+        try {
+            assertNotNull(processInstanceId);
+            assertTrue(processInstanceId.longValue() > 0);
+
+            // Process instance is running and is active.
+            ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ID_RESTART, processInstanceId);
+            assertNotNull(processInstance);
+            assertEquals(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE, processInstance.getState().intValue());
+
+            processClient.abortProcessInstance(CONTAINER_ID_RESTART, processInstanceId);
+
+            List<NodeInstance> list = this.processClient.findNodeInstancesByType(CONTAINER_ID_RESTART, processInstanceId, "ABORTED", 0, 10);
+            String[] nodeIds = list.stream().map(e -> e.getNodeId()).toArray(String[]::new);
+
+            processInstanceId = null;
+            processInstanceId = processClient.startProcessFromNodeIds(CONTAINER_ID_RESTART, PROCESS_ID_RESTART, parameters, nodeIds);
+
+            processClient.abortProcessInstance(CONTAINER_ID_RESTART, processInstanceId);
+        } catch (Exception e) {
+            fail(e.getMessage());
+            if (processInstanceId != null) {
+                processClient.abortProcessInstance(CONTAINER_ID_RESTART, processInstanceId);
+            }
+        }
+    }
+
+    @Test()
+    public void testStartWithCorrelationKeyProcess() {
+        Map<String, Object> parameters = new HashMap<String, Object>();
+
+        Long processInstanceId = processClient.startProcess(CONTAINER_ID_RESTART, PROCESS_ID_RESTART, parameters);
+        try {
+            assertNotNull(processInstanceId);
+            assertTrue(processInstanceId.longValue() > 0);
+
+            // Process instance is running and is active.
+            ProcessInstance processInstance = processClient.getProcessInstance(CONTAINER_ID_RESTART, processInstanceId);
+            assertNotNull(processInstance);
+            assertEquals(org.kie.api.runtime.process.ProcessInstance.STATE_ACTIVE, processInstance.getState().intValue());
+
+            processClient.abortProcessInstance(CONTAINER_ID_RESTART, processInstanceId);
+
+            List<NodeInstance> list = this.processClient.findNodeInstancesByType(CONTAINER_ID_RESTART, processInstanceId, "ABORTED", 0, 10);
+            String[] nodeIds = list.stream().map(e -> e.getNodeId()).toArray(String[]::new);
```

Comment:
Can be replaced with method reference
```suggestion
            String[] nodeIds = list.stream().map(NodeInstance::getNodeId).toArray(String[]::new);
```

Reply:
done